### PR TITLE
Change google maps api key by google maps client id

### LINKED ISSUE #3589: use client_id instead of api_key for google maps
We should use client attribute instead of api_key

https://developers.google.com/maps/documentation/business/clientside/

### PR DESCRIPTION
**Pull request by xavijam**
This changes the way we set google maps key. As of today, google maps key is now google maps client id, and it will be set as in:

 https://developers.google.com/maps/documentation/business/clientside/

screen shot 2015-05-11 at 15 52 52
screen shot 2015-05-11 at 15 53 23

Fixes #3589.
REVIEWER: @javisantana

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3595/))

**Comment by xavijam** [comment]
Take a look now @javisantana when you have time.

**Comment by xavijam** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3626/))

**Comment by xavijam** [comment]
:dancer:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3627/))

**Comment by saleiva** [comment]
bump!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3825/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3826/))

**Comment by javisantana** [comment]
@xavijam review it please

File: app/views/admin/users/account.html.erb [hunk 1]
```diff
@@ -127,12 +127,12 @@
...
                 <p class="Form-rowInfoText Form-rowInfoText--multipleLines">
-                  This is your Google Maps API key,
+                  This is your Google Maps query string,
                   contact with <a href="mailto:support@cartodb.com">support@cartodb.com</a> to change it.
                 </p>
               <% elsif current_user.organization.present? && current_user.organization_owner? %>
                 <p class="Form-rowInfoText Form-rowInfoText--multipleLines">
-                  This is the <%= current_user.organization.name %> Google Maps API key,
+                  This is the <%= current_user.organization.name %> Google Maps Client ID,
                   contact with <a href="mailto:support@cartodb.com">support@cartodb.com</a> to change it.
```
Comment (xavijam) [inline app/views/admin/users/account.html.erb]: Query string as well.

File: app/views/admin/visualizations/public/_embed_map_inline_js.html.erb [hunk 2]
```diff
@@ -1,6 +1,5 @@
 <% if @visualization.map.provider == 'googlemaps' %>
-<script type="text/javascript" src="//maps.google.com/maps/api/js?sensor=false&v=3.12&api_key=<%= @visualization.user.google_maps_api_key %>
-"></script>
+<script type="text/javascript" src="//maps.googleapis.com/maps/api/js?sensor=false&v=3.12&<%= @visualization.user.google_maps_api_key %>></script>
```
Comment (xavijam) [inline app/views/admin/visualizations/public/_embed_map_inline_js.html.erb]: I forgot a ```"``` at the end, please :).

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3827/))

File: app/views/admin/visualizations/show.html.erb [hunk 3]
```diff
@@ -23,7 +23,7 @@
   <% content_for(:js) do %>
   <% if !@google_maps_api_key.blank? %>
         <script type="text/javascript"
-            src="https://maps.google.com/maps/api/js?sensor=false&v=3.12&api_key=<%= @google_maps_api_key %>">
+            src="//maps.googleapis.com/maps/api/js?sensor=false&v=3.12<%= "&client=#{@google_maps_api_key}" if !@google_maps_api_key.blank? %>">
         </script>
```
Comment (xavijam) [inline app/views/admin/visualizations/show.html.erb]: won't client parameter be added using the query string? So let's remove it from here.

File: app/views/admin/visualizations/public_table.html.erb [hunk 4]
```diff
@@ -9,9 +9,9 @@
 
 <% content_for(:js) do %>
     <% if @table.map.provider == 'googlemaps' %>
-      <script type="text/javascript"
-        src="https://maps.google.com/maps/api/js?sensor=false&v=3.12&api_key=<%= @user.google_maps_api_key %>">
```
Comment (xavijam) [inline app/views/admin/visualizations/public_table.html.erb]: We will need the checker, I forgot it. 'blablabla' if true

**Comment by xavijam** [comment]
Basically we should review where we load the google maps library and add the conditional if we need to add the new ```google maps query string```.

**Comment by xavijam** [comment]
Looks good.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3829/))

**Pull request merged**